src/adapter/src/rbac.rs

```diff
@@ -418,6 +448,417 @@ fn ownership_err(
     }
 }
 
+/// The result of this function is a set of tuples of the form
+/// (What object the privilege is on, What privilege is required, Who must possess the privilege).
+///
+/// We use a [`BTreeSet`] because we may generate duplicate privileges while traversing through the
+/// plan and used objects.
+fn generate_required_privileges(
+    catalog: &impl SessionCatalog,
+    plan: &Plan,
+    depends_on: &Vec<GlobalId>,
+    role_id: RoleId,
+) -> BTreeSet<(ObjectId, AclMode, RoleId)> {
+    match plan {
+        Plan::CreateConnection(plan) => {
+            let mut privileges = BTreeSet::from([(
+                plan.name.qualifiers.clone().into(),
+                AclMode::CREATE,
+                role_id,
+            )]);
+            privileges.extend(generate_item_usage_privileges(catalog, depends_on, role_id));
```

Comment:
Why does this need to be here for connections?

Reply:
Connections can use other connections and secrets. `generate_item_usage_privileges` works on secrets, connections, and types. Maybe that's too general? Do you think I should create three different functions so it's more explicit?

Reply:
Nah, I think it's fine as is.